code/modules/antagonists/wizard/equipment/spellbook.dm

```diff
@@ -583,6 +591,40 @@
 	playsound(user, 'sound/magic/mandswap.ogg', 50, 1)
 	return TRUE
 
+/datum/spellbook_entry/summon/magical_access
+	name = "Advent Ritual of Saint Anarchismea"
+	desc = "Seeks the Saint Anarchismea's spirit and summons them onto the station. Access requirement of every electronic in the station will be magically bypassed, but also every ID card gets AA. an ID card without AA will get it again, so latecomers will get the same so that they won't lose their chance to get the blessing of Saint Anarchismea."
+	cost = 2
+	var/static/recursive = FALSE
+	ritual_invocation = "ALADAL DESINARI ODORI'IN IDO'LEX IVLAROYE VOR ANARCHIDMEA OVOR'E SJIENYLE"
+
+/datum/spellbook_entry/summon/magical_access/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
+	SSblackbox.record_feedback("tally", "wizard_spell_learned", 1, name)
+	active = TRUE
+
+	to_chat(user, "<span class='notice'>You have summoned Saint Anarcismea!</span>")
+	playsound(user, 'sound/magic/forcewall.ogg', 50, 1)
+	message_admins("[ADMIN_LOOKUPFLW(user)] cast 'Advent Ritual of Saint Anarchismea' and everyone gets AA from now.")
+	log_game("[key_name(user)] cast 'Advent Ritual of Saint Anarchismea' and everyone gets AA from now.")
+	book.magical_access = TRUE
+	recursive = TRUE
+
+	for(var/mob/living/H in GLOB.player_list)
+		to_chat(H, "<span class='nicegreen'>You feel a holy spirit's blessing... You feel you can go anywhere you want to go.</span>")
+	var/datum/spellbook_entry/summon/magical_access/dummy = new /datum/spellbook_entry/summon/magical_access // this will go recursive even if the book is gone
+	dummy.spell_recursive()
+	return TRUE
+
+/datum/spellbook_entry/summon/magical_access/proc/spell_recursive()
+	if(!recursive)
+		qdel(src)
+		return
+
+	for(var/obj/item/card/id/I in GLOB.id_cards)
+		I.get_magical_access()
+ 	// this happens recursively to help you when your ID card access is wiped by a mean HoP or something somehow
+	addtimer(CALLBACK(src, .proc/spell_recursive), 60 SECONDS)
```

Comment:
This implementation kind of sucks to be honest, this should just make it impossible to change the cards access with a helper method